### PR TITLE
TBRANDS-28 => Story Carousel

### PR DESCRIPTION
**Pull request by brentmiller1973**
## Description

The birth of the Story Carousel Block

## Jira Ticket

- [[TBRANDS-25](https://arcpublishing.atlassian.net/browse/TBRANDS-25)](https://arcpublishing.atlassian.net/browse/TBRANDS-28)

## Something to note:  

While I don't consider this urgent, when you view this, you will see that on initial render, the underlying carousel component does a wonky re render as its figures out the layout.  Thinking we should look into some kind of callback from the carousel component that lets us know it has rendered correctly.  We could then hide the carousel until we get that callback letting us know we can show it.

## Acceptance Criteria

As a PageBuilder Editor user:
I can add a Story Carousel Block to my page and then configure that block to run off a WebSked collection of Composer Stories (ANS feed) by selecting the WebSked collection from within Page Builder Editor. 

I can input the Carousel heading via a manual custom field in Page Builder Editor (optional field).
If there are less than 4 “Story Cards” alert the user that “This feature requires 4 stories to display.”

Display Logic for Headline and Description:

Headline: We want to limit the headline to 2 rows. If the Headline content expands more than 2 rows we want to truncate the Headline and add an ellipses (…) to the end of the second line. 

Description: We want to limit the description to 3 rows. If the Headline content expands more than 3 rows we want to truncate the Headline and add an ellipses (…) to the end of the third line. 

## Test Steps

1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/story-carousel-block,@wpmedia/single-column-layout-block,@wpmedia/category-carousel-block`
3. Open Storybook: `npm run storybook`
4. To run locally, configure your .env file to use the Arc Commerce Sandbox environment.
5. Run local fusion after pulling other PR: `npx fusion start -f -l @wpmedia/story-carousel-block,@wpmedia/single-column-layout-block`
6. Give the block a headline, a value of 2 for headline truncation and a value of 3 for description truncation. ( I think these should be props and not styles as each story carousel may want a different number of lines).
7. Fill block using collection content source....`homepage` is a good enough tag to pull in stories.

## Dependencies or Side Effects

https://github.com/WPMedia/arc-themes-feature-pack/pull/220

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ x] Confirmed all the test steps a reviewer will follow above are working.
- [x ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [ x] Ran `npm run test`, made sure all tests are passing
  - [x ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	return (
+		// <Stack className={getParentClassNames()}>
+		<Stack className={BLOCK_CLASS_NAME}>
+			{headline && headline.trim() !== "" && <Heading>{headline}</Heading>}
+
+			{elements && elements.length && (
+				<Carousel
+					id={identifier}
+					label={carouselLabel}
+					slidesToShow={slidesToShow}
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: Remove this prop and the component will work as you expect it too - This prop will set the slidesToShow for all viewports

Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Perfect!

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { extractImageFromStory } from "@wpmedia/resizer-image-block";
+import { useFusionContext, useComponentContext } from "fusion:context";
+import getProperties from "fusion:properties";
+import getTranslatedPhrases from "fusion:intl";
+
+const BLOCK_CLASS_NAME = "b-story-carousel";
+
+const StoryCarousel = ({
+	customFields: {
+		inheritGlobalContent = false,
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: We don't need to account for inheriting global content is this block

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import PropTypes from "@arc-fusion/prop-types";
+import { Carousel, Image, Icon, Heading, Paragraph, Stack } from "@wpmedia/arc-themes-components";
+import { useContent } from "fusion:content";
+import { extractImageFromStory } from "@wpmedia/resizer-image-block";
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: Blocks should not consume other blocks. We should make this utility available within the `arc-themes-components` package

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+}) => {
+	const { id } = useComponentContext();
+	const { arcSite } = useFusionContext();
+	const { locale = "en" } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+
+	const content = useContent({
+		source: contentService,
+		query: contentConfigValues,
+	});
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: PageBuilder is giving errors around content_elements this is because later in the block you are destructing the object but in some cases there is no object - we should set an OR to a blank object to stop further code erroring

```suggestion
	}) || [];
```

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+			? typeof carouselContentConfig === "undefined"
+			: inheritGlobalContent;
+	const { content_elements: elements = [] } = showGlobalContent ? globalContent : content;
+
+	if (!elements.length) return null;
+
+	return (
+		<Stack className={BLOCK_CLASS_NAME}>
+			{headerText && headerText.trim() !== "" && (
+				<Heading className={`${BLOCK_CLASS_NAME}__main-title`}>{headerText}</Heading>
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: This `<Heading>` will output a `<h1>` tag which isn't desired, the component should be wrapped in a `<HeadingSection>`

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+		typeof inheritGlobalContent === "undefined"
+			? typeof carouselContentConfig === "undefined"
+			: inheritGlobalContent;
+	const { content_elements: elements = [] } = showGlobalContent ? globalContent : content;
+
+	if (!elements.length) return null;
+
+	return (
+		<Stack className={BLOCK_CLASS_NAME}>
+			{headerText && headerText.trim() !== "" && (
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: We should be using ternary to do rendering logic 

```
{headerText && headerText.trim() !== "" ? (
    <Heading className={`${BLOCK_CLASS_NAME}__main-title`}>{headerText}</Heading>
) : null}
```

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+				<Heading className={`${BLOCK_CLASS_NAME}__main-title`}>{headerText}</Heading>
+			)}
+
+			{elements && elements.length && (
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: Ternary operator should be used to do rendering

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+							key={item._id}
+							label={`${phrases.t("story-carousel.slide-indicator", {
+								current: index + 1,
+								maximum: elements.length,
+							})}`}
+						>
+							<a className={`${BLOCK_CLASS_NAME}__story-card`} href={item.canonical_url}>
+								<Image alt={item.headlines.basic} src={extractImageFromStory(item)} />
+								{item.headlines?.basic ? (
+									<Heading
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: `<Heading>` here requires the `<Carousel>` to be wrapped in a `<HeadingSection>` - but **note** we only want to wrap the carousel in a HeadingSection if the block has a headline. We want to ensure we increase the heading level hierarchy between the block heading and the heading used within the carousel

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+					type: "image",
+					url: testImageURL,
+				},
+			},
+		},
+	],
+	website: "arc-commerce",
+};
+
+const mockPhrases = {
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.test.jsx]: We should not be mocking translations

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+	"global.story-carousel.aria-label": "Stories",
+	"global.story-carousel.right-arrow-label": "Next",
+	"global.story-carousel.left-arrow-label": "Previous",
+	"global.story-carousel.slide-indicator": "Slide %{current} of %{maximum}",
+};
+
+jest.mock("fusion:intl", () => ({
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.test.jsx]: We should not be mocking this library here - there is a global mock that handles intl for us

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+	__esModule: true,
+	default: jest.fn(() => ({ t: jest.fn((phrase) => mockPhrases[phrase]) })),
+}));
+
+jest.mock("fusion:properties", () =>
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.test.jsx]: This mock should not be needed as we are not using any of these values

File: blocks/story-carousel-block/package.json [hunk 12]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@wpmedia/story-carousel-block",
+  "version": "0.1.0",
+  "description": "Hero Block",
+  "homepage": "https://github.com/WPMedia/arc-themes-blocks",
+  "license": "CC-BY-NC-ND-4.0",
+  "main": "",
+  "files": [
+    "_index.scss",
```
Comment (matthewroach) [inline blocks/story-carousel-block/package.json:9]: Need to add `intl.json` to the files array to ensure the file is published

File: blocks/story-carousel-block/package.json [hunk 13]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@wpmedia/story-carousel-block",
+  "version": "0.1.0",
+  "description": "Hero Block",
```
Comment (matthewroach) [inline blocks/story-carousel-block/package.json]:
```suggestion
  "description": "Story Carousel Block",
```

File: blocks/story-carousel-block/index.story.jsx [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+import StoryCarousel from "./features/story-carousel/default";
+
+export default {
+	title: "Blocks/Story Carousel",
+	parameters: {
+		chromatic: { viewports: [320, 1200] },
+	},
+};
+
+const testImageURL =
```
Comment (matthewroach) [inline blocks/story-carousel-block/index.story.jsx]: We have images available to use from the `/resources` directory - we can use those rather than having an inlined base64 image

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+		inheritGlobalContent: PropTypes.bool.tag({
+			group: "Configure Content",
+			defaultValue: false,
+		}),
+		headerText: PropTypes.string.tag({
+			label: "Headline",
+			default: "",
+			description: "Add a headline for the story carousel component.",
+		}),
+		itemHeaderTruncationLines: PropTypes.number.tag({
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: The truncation is handled via the theme - we don't need to expose the truncation options via custom fields

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+								current: index + 1,
+								maximum: elements.length,
+							})}`}
+						>
+							<a className={`${BLOCK_CLASS_NAME}__story-card`} href={item.canonical_url}>
+								<Image alt={item.headlines.basic} src={extractImageFromStory(item)} />
+								{item.headlines?.basic ? (
+									<Heading
+										className={`${BLOCK_CLASS_NAME}__story-card-header`}
+										truncationLines={itemHeaderTruncationLines}
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: The truncation is handled via the theme - we don't need to expose the truncation options via custom fields and that would remove the need to add additional testing for an already tested component

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+										data-testid={itemHeaderTruncationLines}
+									>
+										{item.headlines.basic}
+									</Heading>
+								) : null}
+								{item.description?.basic ? (
+									<Paragraph
+										truncationLines={itemDescriptionTruncationLines}
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: Same comment around truncation

> The truncation is handled via the theme - we don't need to expose the truncation options via custom fields and that would remove the need to add additional testing for an already tested component

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+	const content = useContent({
+		source: contentService,
+		query: contentConfigValues,
+	});
+
+	const showGlobalContent =
+		typeof inheritGlobalContent === "undefined"
+			? typeof carouselContentConfig === "undefined"
+			: inheritGlobalContent;
+	const { content_elements: elements = [] } = showGlobalContent ? globalContent : content;
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: We don't need to account for inheriting global content is this block

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+					{elements.map((item, index) => (
+						<Carousel.Item
+							key={item._id}
+							label={`${phrases.t("story-carousel.slide-indicator", {
+								current: index + 1,
+								maximum: elements.length,
+							})}`}
+						>
+							<a className={`${BLOCK_CLASS_NAME}__story-card`} href={item.canonical_url}>
+								<Image alt={item.headlines.basic} src={extractImageFromStory(item)} />
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: alt should be a blank string is if there is a heading or description otherwise there will be duplicated text for assistive technologies

**Review by matthewroach — CHANGES_REQUESTED** [review]

**Comment by matthewroach** [comment]
The header is not always aligned to the left edge as per the designs - 

Microsoft Edge-2022-04-22-12-02 24

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+		itemDescriptionTruncationLines = 0,
+	},
+	globalContent = {},
+}) => {
+	const { id } = useComponentContext();
+	const { arcSite } = useFusionContext();
+	const { locale = "en" } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+
+	const content = useContent({
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: We should make use of the filter param here to ensure we are only getting the data we need, we also need to specify an additional query param to avoid caching issues when using `filter` as per this example - https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-1.21/blocks/small-promo-block/features/small-promo/default.jsx#L15

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 6 above)
Comment (RobinGiannattasio) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Why is the ternary considered better? I do it and know we have it as a norm, but I prefer the simplicity of the `&&` for readability. TEACH ME!

Comment (JackHowa) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: this is the best explanation I've seen! https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 2 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Yup I agree.....but I added it so I could get storybook to work.  Could not figure out how to do a story without injecting global content.  Any ideas @matthewroach ?

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 6 above)
Comment (RobinGiannattasio) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Thank you!

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 18 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: See previous comment @matthewroach about storybook support.  I need guidance with this.

Comment (JackHowa) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: if you look at https://github.com/WPMedia/arc-themes-blocks/tree/arc-themes-release-version-1.21/.storybook/mock-content https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-1.21/.storybook/alias/content.js#L12 you can see how content is mocked from useContent. if you want to pair, feel free to reach out!

File: .storybook/themes/commerce.scss [hunk 21]
```diff
@@ -329,6 +329,61 @@
...
+						carousel-button: (
+							border-radius: var(--global-spacing-1),
+							background-color: var(--global-white),
+							box-shadow: 0px 0px 16px 0px #19191966,
+							padding: 0.75em,
+						),
+						carousel-track: (
+							gap: var(--global-spacing-5),
+							max-width: 84.7%,
+						),
```
Comment (nschubach) [inline .storybook/themes/commerce.scss]: Just an FYI, these are now included in the default/components so you should probably not need them.

The removal of top margin is necessary?

File: .storybook/themes/commerce.scss [hunk 22]
```diff
@@ -390,10 +445,36 @@
...
+							--slides: 4,
+						),
+						carousel-actions: (
+							display: flex,
+							margin-top: -7rem,
+						),
+						carousel-track: (
+							justify-self: center,
+							max-width: 88%,
+						),
```
Comment (nschubach) [inline .storybook/themes/commerce.scss]: Also in desktop/components, except the margin shift.  Why are we applying negative margin (and so much of it)?...

File: .storybook/themes/commerce.scss [hunk 23]
```diff
@@ -390,10 +445,36 @@
...
+				),
+				category-carousel-title: (
+					components: (
+						heading: (
+							font-size: var(--global-font-size-7),
+							line-height: var(--global-line-height-4),
+							font-weight: var(--global-font-weight-7),
+						),
+					),
+					padding-left: 6rem,
```
Comment (nschubach) [inline .storybook/themes/commerce.scss]: If you set the width of the header on default to `var(--layout-content-scale-width)` with auto margins it should match the track given the new defaults.

File: .storybook/themes/commerce.scss [hunk 24]
```diff
@@ -390,10 +445,36 @@
...
+							display: flex,
+							margin-top: -7rem,
+						),
+						carousel-track: (
+							justify-self: center,
+							max-width: 88%,
+						),
+					),
+				),
+				category-carousel-title: (
```
Comment (nschubach) [inline .storybook/themes/commerce.scss]:
```suggestion
				story-carousel-title: (
```

File: .storybook/themes/commerce.scss [hunk 25]
```diff
@@ -329,6 +329,61 @@
...
+							max-width: 84.7%,
+						),
+						heading: (
+							font-size: var(--global-font-size-11),
+							line-height: var(--global-line-height-6),
+						),
+					),
+					gap: 0.5rem,
+				),
+				category-carousel-title: (
```
Comment (nschubach) [inline .storybook/themes/commerce.scss]:
```suggestion
				story-carousel-title: (
```

File: blocks/story-carousel-block/_index.scss [hunk 26]
```diff
@@ -0,0 +1,25 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-story-carousel {
+	&__main-title {
+		@include scss.block-properties("category-carousel-title");
+		@include scss.block-components("category-carousel-title");
```
Comment (nschubach) [inline blocks/story-carousel-block/_index.scss]:
```suggestion
		@include scss.block-properties("story-carousel-title");
		@include scss.block-components("story-carousel-title");
```

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 27]
```diff
@@ -0,0 +1,131 @@
...
+		carouselContentConfig: { contentService = "", contentConfigValues = {} } = {},
+		headerText,
+		itemHeaderTruncationLines = 0,
+		itemDescriptionTruncationLines = 0,
+	},
+	globalContent = {},
+}) => {
+	const { id } = useComponentContext();
+	const { arcSite } = useFusionContext();
+	const { locale = "en" } = getProperties(arcSite);
```
Comment (JackHowa) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]:
```suggestion
	const { locale } = getProperties(arcSite);
```

don't need the 'en' fallback

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 28]
```diff
@@ -0,0 +1,131 @@
...
+			default: 0,
+			description:
+				"Number of lines the description in a carousel item will show before being truncated",
+		}),
+	}),
+};
+
+StoryCarousel.label = "Story Carousel - Arc Block";
+
+StoryCarousel.icon = "ui-browser-slider";
```
Comment (JackHowa) [inline blocks/story-carousel-block/features/story-carousel/default.jsx:154]: was this slider the pick from product for the icon? not sure if this was in the ticket

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 2 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 3 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: That has been taken care of.

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 4 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 5 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: wrapped

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 6 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: got it

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 7 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: got it

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 8 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: got it

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 15 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 16 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 17 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: yup done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 18 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 19 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: Done

**Comment by brentmiller1973** [comment]
> The header is not always aligned to the left edge as per the designs -
> 
> Microsoft Edge-2022-04-22-12-02 24

Should be now

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 28 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx:154]: same as the content carousel

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 27 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: ok done

File: blocks/story-carousel-block/_index.scss (hunk 26 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/_index.scss]: Haha you know what I was doing now ;-). Done

File: .storybook/themes/commerce.scss (hunk 23 above)
Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: I think I got it

File: .storybook/themes/commerce.scss (hunk 24 above)
Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: Yup got it

File: .storybook/themes/commerce.scss (hunk 25 above)
Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: done

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 28 above)
Comment (JackHowa) [reply blocks/story-carousel-block/features/story-carousel/default.jsx:154]: cool

File: blocks/story-carousel-block/README.md [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+| ------------ | ------------ | --------------- | ------------------------- |
+| customFields | yes          | PropTypes.shape | Pagebuilder Custom Fields |
+
+### customFields
+
+| **Prop**                       | **Required** | **Type**         | **Description**                                                         |
+| ------------------------------ | ------------ | ---------------- | ----------------------------------------------------------------------- |
+| headerText                     | no           | PropTypes.string | Carousel Header text                                                    |
+| itemHeaderTruncationLines      | no           | PropTypes.string | Number of lines to show in the card header before being truncated.      |
+| itemDescriptionTruncationLines | no           | PropTypes.string | Number of lines to show in the card description before being truncated. |
```
Comment (JackHowa) [inline blocks/story-carousel-block/README.md]: can you update the truncated fields based on removing this from the custom field available options 
```suggestion
```

**Review by JackHowa — DISMISSED** [review]
Good work addressing the syntax issues, like ternaries. I know it can not seem like a lot but gotta watch for those edge cases. Looks like the tests in the cloud are failing in the github action @visionbegin. Did you run `npm i` then `git add` the package-lock and push to get the package-lock.json change? 

It also looks like there's existing comments that haven't been addressed in the block package.json and styles.

File: blocks/story-carousel-block/features/story-carousel/default.jsx [hunk 30]
```diff
@@ -0,0 +1,154 @@
...
+    }`,
+		}) || [];
+
+	const { content_elements: elements = [] } = content;
+	if (!elements.length) return null;
+
+	return (
+		<Stack className={BLOCK_CLASS_NAME}>
+			{headerText && headerText.trim() !== "" ? (
+				<HeadingSection>
```
Comment (matthewroach) [inline blocks/story-carousel-block/features/story-carousel/default.jsx]: See this PR - https://github.com/WPMedia/arc-themes-blocks/pull/1320 for an example of how the `<HeadingSection>` wrapping of the block should be

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 30 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: I added it.

File: blocks/story-carousel-block/README.md (hunk 29 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/README.md]: Updated

File: blocks/story-carousel-block/features/story-carousel/default.jsx (hunk 20 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.jsx]: filter has been added

File: .storybook/themes/commerce.scss (hunk 21 above)
Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: Reduced my own styles somewhat to just what I needed.  Yes top margin is needed to pick up the corresponding desktop breakpoint

File: .storybook/themes/commerce.scss (hunk 22 above)
Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: To move the buttons up more to be centered with the images and not the whole card itself.

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx (hunk 9 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.test.jsx]: removed

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx (hunk 10 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.test.jsx]: removed

File: blocks/story-carousel-block/features/story-carousel/default.test.jsx (hunk 11 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/features/story-carousel/default.test.jsx]: removed

File: blocks/story-carousel-block/package.json (hunk 12 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/package.json:9]: added

File: blocks/story-carousel-block/package.json (hunk 13 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/package.json]: updated

File: blocks/story-carousel-block/index.story.jsx (hunk 14 above)
Comment (brentmiller1973) [reply blocks/story-carousel-block/index.story.jsx]: updated

File: .storybook/themes/commerce.scss [hunk 31]
```diff
@@ -396,6 +396,54 @@
...
+				story-carousel: (
+					components: (
+						carousel: (
+							--slides: 1,
+						),
+						carousel-actions: (
+							margin-top: 0,
+							// Need to have the setting here so desktop gets included
+						),
+						carousel-track: (
```
Comment (matthewroach) [inline .storybook/themes/commerce.scss]: Once feedback is addressed in the feature pack PR we can bring those changes over to here

Comment (brentmiller1973) [reply .storybook/themes/commerce.scss]: @matthewroach Thank you for this, I have integrating the change in both the themepack and in the block repo.  Confirmed in both Fusion and Storybook.

**Review by matthewroach — APPROVED** [review]

**Review by matthewroach — APPROVED** [review]
